tests/foreman/maintain/test_packages.py

```diff
@@ -0,0 +1,293 @@
+    assert 'FAIL' not in result.stdout
+    assert 'Packages are not locked.' in result.stdout
+    assert 'Automatic locking of package versions is disabled in installer.' in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Packages.is_locked()
+    assert 'Packages are not locked' in result.stdout
+    assert result.status == 1
+
+    # Teardown: lock packages of satellite, for capsule its unlocked by default
+    if sat_maintain.__class__.__name__ == 'Satellite':
+        result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+        assert result.status == 0
+        assert 'Success!' in result.stdout
+
+
+@pytest.mark.include_capsule
+def test_positive_fm_packages_install(request, sat_maintain):
+    """Verify whether packages install/update work as expected.
+
+    :id: 645a3d84-34cb-469c-8b79-105b889aac78
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-installer --lock-package-versions
+        2. Run satellite-maintain packages status
+        3. Run satellite-maintain packages is-locked
+        4. Try to install/update package using satellite-maintain packages install/update command.
+        5. Run satellite-installer --no-lock-package-versions
+        6. Run satellite-maintain packages status
+        7. Run satellite-maintain packages is-locked
+        8. Try to install package in unlocked state.
+        9. Teardown (Run satellite-installer --lock-package-versions)
+
+    :expectedresults: Packages get install/update when lock/unlocked.
+    """
+    # Test whether packages are locked or not
+    result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+    assert result.status == 0
+    assert 'Success!' in result.stdout
+
+    result = sat_maintain.cli.Packages.status()
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Packages.is_locked()
+    assert 'Packages are locked' in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.execute('dnf install -y zsh')
+    assert result.status == 1
+    assert 'Use foreman-maintain packages install/update <package>' in result.stdout
+
+    # Test whether satellite-maintain packages install/update command works as expected.
+    result = sat_maintain.cli.Packages.install(packages='zsh', options={'assumeyes': True})
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'Nothing to do' not in result.stdout
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+
+    result = sat_maintain.cli.Packages.update(packages='zsh', options={'assumeyes': True})
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'Nothing to do' not in result.stdout
```

Comment:
I am not sure this is working because the step is installing zsh with the latest version and how we could expect it would update again !

This step should be 
```suggestion
    assert 'Nothing to do' in result.stdout
```

Reply:
if packages are locked we nothing to do in stdout, but here we use f-m to install package, so it works and we shouldn't get nothing to do, so this assertion is valid